js/src/common/components/Button.tsx

```diff
@@ -66,7 +66,7 @@ export interface IButtonAttrs extends ComponentAttrs {
  * be used to represent any generic clickable control, like a menu item. Common
  * styles can be applied by providing `className="Button"` to the Button component.
  */
-export default class Button extends Component<IButtonAttrs> {
+export default class Button<CustomAttrs extends Record<string, unknown> = {}> extends Component<IButtonAttrs & CustomAttrs> {
```

Comment:
Why can't we extend ComponentAttrs

Reply:
We can, there's just no point in doing so. Including `ComponentAttrs` is done with the `IButtonAttrs` interface.

Reply:
Can't the same thing be achieved by doing `CustomAttrs extends IButtonAttrs`? I think that's what is being done in other components?

Reply:
It can be done that way, but that runs the risk of depending on the component extending it to correctly extend the `IButtonAttrs` interface. I'd prefer to enforce IButtonAttrs as being present.

Reply:
Nope. Screw that. Changed my mind. Totally agree with you.
```suggestion
export default class Button<CustomAttrs extends IButtonAttrs = IButtonAttrs> extends Component<CustomAttrs> {
```